test/sql-luatest/view_test.lua

```diff
@@ -0,0 +1,467 @@
+        t.assert(box.space.b ~= nil)
+        box.execute("CREATE TABLE c (s1 INT PRIMARY KEY);")
+        t.assert(box.space.c ~= nil)
+        box.execute([[CREATE VIEW bcv AS SELECT * FROM b
+                                     WHERE s1 IN (SELECT * FROM c);]])
+        t.assert(box.space.bcv ~= nil)
+        _, err = box.execute("DROP TABLE c;")
+        error = "Can't drop space 'c': other views depend on this space"
+        t.assert_equals(err.message, error)
+        box.execute("DROP VIEW bcv;")
+        box.execute("DROP TABLE c;")
+
+        box.execute("CREATE TABLE c (s1 INT PRIMARY KEY);")
+        t.assert(box.space.c ~= nil)
+        box.execute([[CREATE VIEW bcv(x, y) AS VALUES((SELECT 'k' FROM b),
+                     (VALUES((SELECT 1 FROM b WHERE s1 IN
+                     (VALUES((SELECT 1 + c.s1 FROM c)))))));]])
+        t.assert(box.space.bcv ~= nil)
+        _, err = box.execute("DROP TABLE c;")
+        error = "Can't drop space 'c': other views depend on this space"
+        t.assert_equals(err.message, error)
+        box.space.bcv:drop()
+        box.execute("DROP TABLE c;")
+        box.execute("DROP TABLE b;")
+    end)
+end
+
+-- gh-3814: make sure that recovery of view processed without
+-- unexpected errors.
+--
+g.test_3814_view_recovery_check = function()
+    g.server:exec(function()
+        box.snapshot()
+        box.execute("CREATE TABLE t1 (id INT PRIMARY KEY);")
+        t.assert(box.space.t1 ~= nil)
+        box.execute("CREATE VIEW v1 AS SELECT * FROM t1;")
+        t.assert(box.space.v1 ~= nil)
+
+        local _, err = box.execute("DROP TABLE t1;")
+        local error = "Can't drop space 't1': other views depend on this space"
+        t.assert_equals(err.message, error)
+        local exp = {
+            metadata = {
+                {name = "id", type = "integer"}
+            },
+            rows = {},
+        }
+        box.execute("SELECT * FROM v1;")
+        box.space.v1:drop()
+        box.space.t1:drop()
+    end)
+end
+
+--
+-- gh-4740: make sure INSTEAD OF DELETE and INSTEAD OF UPDATE
+-- triggers work for each row of view.
+--
+g.test_4740_instead_of_delete_update_check = function()
+    g.server:exec(function()
+        box.execute('CREATE TABLE t (i INT PRIMARY KEY AUTOINCREMENT, a INT);')
+        t.assert(box.space.t ~= nil)
+        box.execute('CREATE TABLE t1 (i INT PRIMARY KEY AUTOINCREMENT, a INT);')
+        t.assert(box.space.t1 ~= nil)
+        box.execute('CREATE VIEW v AS SELECT a FROM t;')
+        t.assert(box.space.v ~= nil)
+        box.execute([[CREATE TRIGGER r1 INSTEAD OF DELETE ON v
+                     FOR EACH ROW BEGIN INSERT INTO t1 VALUES (NULL, 1); END;]])
+        box.execute([[CREATE TRIGGER r2 INSTEAD OF UPDATE ON v
+                     FOR EACH ROW BEGIN INSERT INTO t1 VALUES (NULL, 2); END;]])
+        local exp = {
+            autoincrement_ids = {1, 2, 3, 4, 5, 6},
+            row_count = 6
```

Comment:
Let's add a comma when a table spans multiple lines. For example:
```Lua
local a = {
    qwe = 1,
    asd = 2,
    zxc = 3,
}
```

If the table is written in a row, a comma is not needed:
```Lua
local a = {qwe = 1, asd = 2, zxc = 3}
```